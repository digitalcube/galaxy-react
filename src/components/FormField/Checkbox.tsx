import React , {FC} from 'react'
import { FormGroup, Input, InputProps, Label } from 'reactstrap'


export type CheckboxProps = {
    label: string;
    id: string;
} & Omit<InputProps, 'type'>
export const Checkbox: FC<CheckboxProps> = (props) => {
    return (
        <FormGroup className="form-check d-flex align-items-center colored-check smaller">
            <Input {...props} type="checkbox" id={props.id} className="screen-reader-text" checked={props.value === true} />
            <Label className="form-check-label" htmlFor={props.id}>{props.label}</Label>
        </FormGroup>
    )
}