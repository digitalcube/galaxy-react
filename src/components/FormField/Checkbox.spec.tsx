import { render } from '@testing-library/react';
import React from 'react';
import { Default as Checkbox } from './Checkbox.stories';

describe('Checkbox', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <Checkbox {...(Checkbox.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
