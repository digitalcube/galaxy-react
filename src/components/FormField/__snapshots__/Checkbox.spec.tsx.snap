// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-check d-flex align-items-center colored-check smaller form-group"
  >
    <input
      class="screen-reader-text form-check-input"
      id="remember"
      label="Checkbox name"
      type="checkbox"
      value="false"
    />
    <label
      class="form-check-label"
      for="remember"
    >
      Checkbox name
    </label>
  </div>
</DocumentFragment>
`;
